### DISCUSSION TITLE
Activity log

### DISCUSSION THREAD

**Discussion by egrosuero** [gitextensions/gitextensions#11399] [General]
Hi All.

First of all, I´m not developer and I have never used Git.

I should deal with an issue with Gitextensions wich I need to obtain an "activity log" from the Gitextensios in order to determine when a branch was deleted by the local Gitextension´s user.

Is Gitextensions mantain  some kind of log that could help me to determine it? If so, where can I found it?

If not, what I should search for find information about the local developer activity?.

Thanks in advance to everybody.

Best regards.

Enrique.

**Comment by pmiossec** [comment]
No, neither GitExtensions nor Git keep this kind of information.

If you **really** need that, you should rely on the filesystem logging looking for deletion of file `.git/refs/heads/NAME_OF_YOUR_BRANCH` (for a local branch).
But I don't think you will have these information either except if your company enabled some logging before.

Is there a specific reason to try to get this information? For a local branch or a remote one?
Maybe try to find the last dangling commits belonging to this branch (in menu `Repository=> Git maintenance => Recover lost objects...` and you will have an idea of when it happened (after the last commit you get and before the one your don't have).

**Reply by egrosuero** [reply]
Hi pmiossec.

Thank you for your kindly response.

_But I don't think you will have these information either except if your company enabled some logging before._

It is the case. Recently we enable the O.S. logging for filesystem access, but when the event occuried, this was not enabled.

_Is there a specific reason to try to get this information? For a local branch or a remote one?_

The reason is a malicious behavior from a worker. He deleted some code branches on the remote repository.  We restore the code from server backups and the management has release a forensic investigation for take legal actions.

We are exploring the command

git log --author="author_name"

... but we are not sure if could be possible to obtain valuable information once we recovered the deleted branches from the backups.

Thanks again for your help.

Best regards.

**Reply by RussKie** [reply]
Git Extensions is a UI over git cli. The app isn't capturing any logs outside the git commands log (<kbd>F12</kbd>) _while_ the app is running.

**Reply by vbjay** [reply]
You can also look at the reflog. https://stackoverflow.com/questions/3640764/can-i-recover-a-branch-after-its-deletion-in-git
